sigopt/xgboost/experiment.py

```diff
@@ -9,13 +9,24 @@
   DEFAULT_REGRESSION_METRIC,
   DEFAULT_SEARCH_PARAMS,
   METRICS_OPTIMIZATION_STRATEGY,
+  PARAMETER_INFORMATION,
   SEARCH_BOUNDS,
-  SEARCH_PARAMS,
+  SUPPORTED_AUTOBOUND_PARAMS,
   SUPPORTED_METRICS_TO_OPTIMIZE,
 )
 from .run import parse_run_options
 from .run import run as XGBRunWrapper
 
+
+def check_if_bounds_in_limits(bounds, limits):  #TODO move to utils, don't want to deal with merge conflict atm
+  min_limit_type = limits[0]
+  max_limit_type = limits[-1]
+  min_limit, max_limit = [float(f) for f in limits[1:-1].split(',')]
+  IN_BOUNDS_MIN = min_limit < bounds['min'] if min_limit_type == '(' else min_limit <= bounds['min']
+  IN_BOUNDS_MAX = max_limit > bounds['min'] if max_limit_type == ')' else max_limit >= bounds['min']
+  return all([IN_BOUNDS_MIN, IN_BOUNDS_MAX])
```

Comment:
lol, this is kinda ugly. We have this `Interval` class in `sigopt-api` that maybe we can use here. Just a thought.

Reply:
I prefer create a function to parse PARAMETER_INFORMATION to a dict of Interval.

Reply:
hmm copying over the interval class seems like more code than needed. Though I agree that this is some ugly code lol